Frameworks/AddressBook/ABRecord.mm

```diff
@@ -47,12 +82,118 @@ bool ABRecordSetValue(ABRecordRef record, ABPropertyID property, CFTypeRef value
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Caveat
+ @Notes Properties that return a Multi-Value are not yet supported, and a few properties
+        (alternate birthday, creation/modification dates, person kind, phonetic middle name)
+        don't have direct equivalents.
 */
-CFTypeRef ABRecordCopyValue(ABRecordRef record, ABPropertyID property) {
-    UNIMPLEMENTED();
-    return StubReturn();
+CFTypeRef ABRecordCopyValue(ABRecordRef record, ABPropertyID contactProperty) {
+    if (record == nullptr) {
+        return nullptr;
+    }
+
+    _ABContact* person = (__bridge _ABContact*)record;
+
+    // Cases for various name properties.
+    if (contactProperty == kABPersonFirstNameProperty) {
+        return (__bridge_retained CFTypeRef)person.contact.firstName;
+    } else if (contactProperty == kABPersonLastNameProperty) {
+        return (__bridge_retained CFTypeRef)person.contact.lastName;
+    } else if (contactProperty == kABPersonMiddleNameProperty) {
+        return (__bridge_retained CFTypeRef)person.contact.middleName;
+    } else if (contactProperty == kABPersonPrefixProperty) {
+        return (__bridge_retained CFTypeRef)person.contact.honorificNamePrefix;
+    } else if (contactProperty == kABPersonSuffixProperty) {
+        return (__bridge_retained CFTypeRef)person.contact.honorificNameSuffix;
+    } else if (contactProperty == kABPersonNicknameProperty) {
+        return (__bridge_retained CFTypeRef)person.contact.nickname;
+    } else if (contactProperty == kABPersonFirstNamePhoneticProperty) {
+        return (__bridge_retained CFTypeRef)person.contact.yomiGivenName;
+    } else if (contactProperty == kABPersonLastNamePhoneticProperty) {
+        return (__bridge_retained CFTypeRef)person.contact.yomiFamilyName;
+    }
+
+    // Cases for job-related properties.
+    if (contactProperty == kABPersonOrganizationProperty) {
+        NSArray* jobInfo = person.contact.jobInfo;
+        return [jobInfo count] > 0 ? (__bridge_retained CFTypeRef)((WACContactJobInfo*)jobInfo[0]).companyName : nullptr;
+    } else if (contactProperty == kABPersonJobTitleProperty) {
+        NSArray* jobInfo = person.contact.jobInfo;
+        return [jobInfo count] > 0 ? (__bridge_retained CFTypeRef)((WACContactJobInfo*)jobInfo[0]).title : nullptr;
+    } else if (contactProperty == kABPersonDepartmentProperty) {
+        NSArray* jobInfo = person.contact.jobInfo;
+        return [jobInfo count] > 0 ? (__bridge_retained CFTypeRef)((WACContactJobInfo*)jobInfo[0]).department : nullptr;
+    }
+
+    // Case for birthday-related property.
```

Comment:
not the most super useful of comments. Please add details on the why of the code rather than just the what, especially for the ones that have significant business logic in the transformation.